elytron/src/main/java/org/wildfly/extension/elytron/ElytronPersistentResourceXMLDescriptionFactory.java

```diff
@@ -0,0 +1,230 @@
+                    .addAttribute(ElytronDefinition.DEFAULT_SSL_CONTEXT);
+        }
+
+        addAuthenticationClientParser(builder, schema);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(new ProviderParser().parser);
+        }
+
+        addAuditLoginParser(builder, schema);
+        addSecurityDomainParser(schema, builder);
+        addRealmParser(schema, builder);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(new CredentialSecurityFactoryParser().parser);
+        }
+
+        addMapperParser(schema, builder);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_3_0)) {
+            builder.addChild(getPermissionSetParser());
+        }
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(new HttpParser().parser);
+        }
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(new SaslParser().parser);
+        }
+
+        addTlsParser(schema, builder);
+        addCredentialStoreParser(schema, builder);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_13_0)) {
+            builder.addChild(getExpressionResolverParser());
+        }
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_3_0)) {
+            builder.addChild(getDirContextParser());
+        }
+
+        addPolicyParser(schema, builder);
+        if (schema.since(ElytronSubsystemSchema.VERSION_5_0)) {
+            builder.addChild(new JaspiConfigurationParser().jaspiConfigurationParser_5_0);
+        }
+
+        return builder.build();
+    }
+
+    private static void addPolicyParser(ElytronSubsystemSchema schema, PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder) {
+        PolicyParser policyParser = new PolicyParser();
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_2)) {
+            builder.addChild(policyParser.parser_1_2);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_1_1)) {
+            builder.addChild(policyParser.parser_1_0);
+        }
+    }
+
+    private static void addCredentialStoreParser(ElytronSubsystemSchema schema, PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder) {
+        CredentialStoreParser credentialStoreParser = new CredentialStoreParser();
+        if (schema.since(ElytronSubsystemSchema.VERSION_13_0)) {
+            builder.addChild(credentialStoreParser.getCredentialStoresParser_13().build());
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(credentialStoreParser.getCredentialStoresParser().build());
+        }
+    }
+
+    private static void addTlsParser(ElytronSubsystemSchema schema, PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder) {
+        TlsParser tlsParser = new TlsParser();
+        if (schema.since(ElytronSubsystemSchema.VERSION_18_0_COMMUNITY)) {
```

Comment:
N.B. ElytronSubsystemSchema.since(...) only compares version numbers.  If a resource needs to be conditionally added based on a specific min version and stability, you would use something like:

```java
if (this.since(VERSION_X_Y) && this.enables(ResourceRegistration.FOO)) {
   // This community-level resource was added at VERSION_X_Y
   builder.addChild(ResourceRegistration.FOO);
}
```

Otherwise, the parser will attempt to persist this resource in the next version regardless of stability level.

Reply:
This is now fixed